### DISCUSSION TITLE
why “blocks of trees”？

### DISCUSSION THREAD

**Discussion by A-Pai** [kogalur/randomForestSRC#162] [Q&A] [answered]
why “blocks of trees”？instead of using all the tree
![image](https://user-images.githubusercontent.com/5112397/148953017-64e1de73-a3f9-41f2-be33-c655c07cbecf.png)
![image](https://user-images.githubusercontent.com/5112397/148953524-27f3a5e3-16cf-4e93-ba6e-08caf3015513.png)

**Comment by ishwaran** [comment, marked as answer]
`block.size=1 ` will use all the trees, however this may be computationally slow, therefore it is sometimes advantageous computationally to increase this value.  This is similar to the usual VIMP (variable importance) which also has this option.

**Reply by A-Pai** [reply]
The functon "randomForest" and "importance" in package randomForest does have argument "block.size"，does this indicate that it is not functional or thoughtless？

**Reply by A-Pai** [reply]
It seems package “ranger”does not  have argument "block.size" either
